tests/python/test_usdt.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+#
+# USAGE: test_usdt.py
+#
+# Copyright 2017 Facebook, Inc
+# Licensed under the Apache License, Version 2.0 (the "License")
+
+from __future__ import print_function
+from bcc import BPF, USDT
+from unittest import main, TestCase
+from subprocess import Popen, PIPE
+from tempfile import NamedTemporaryFile
+import os
+import signal
+
+class TestUDST(TestCase):
+    def setUp(self):
+        # Application, minimum, to define three trace points
+        app_text = """
+#include <unistd.h>
+#include <folly/tracing/StaticTracepoint.h>
+
+int main() {
+  char s[100];
+  int i, a = 20, b = 40;
+  for (i = 0; i < 100; i++) s[i] = (i & 7) + (i & 6);
+  while (1) {
+    FOLLY_SDT(test, probe_point_1, s[7], b);
+    FOLLY_SDT(test, probe_point_3, a, b);
+    sleep(3);
+    a++; b++;
+    FOLLY_SDT(test, probe_point_1, s[4], a);
+    FOLLY_SDT(test, probe_point_2, 5, s[10]);
+    FOLLY_SDT(test, probe_point_3, s[4], s[7]);
+  }
+  return 1;
+}
+"""
+        # BPF program
+        self.bpf_text = """
+#include <uapi/linux/ptrace.h>
+int do_trace(struct pt_regs *ctx) {
+    return 0;
+};
+int do_trace2(struct pt_regs *ctx) {
+    return 0;
+}
+int do_trace3(struct pt_regs *ctx) {
+    return 0;
+}
+"""
+
+        # Compile and run the application
+        self.ftemp = NamedTemporaryFile(delete=False)
+        self.ftemp.close()
+        comp = Popen(["gcc", "-I", "%s/include" % os.getcwd(),
+                      "-x", "c", "-o", self.ftemp.name, "-"],
+                     stdin=PIPE, stdout=PIPE, stderr=PIPE)
+        comp.stdin.write(app_text)
+        comp.stdin.close()
+        self.assertEqual(comp.wait(), 0)
+        self.app = Popen([self.ftemp.name])
+
+    def test_attach1(self):
+        # enable USDT probe from given PID and verifier generated BPF programs
+        u = USDT(pid=int(self.app.pid))
+        u.enable_probe(probe="probe_point_1", fn_name="do_trace")
```

Comment:
This is probably preexisting, but I see a bunch of warnings from the generated code:

```
29: /virtual/main.c:5:26: warning: incompatible integer to pointer conversion assigning to 'void *' from 'volatile int8_t' (aka 'volatile signed char') [-Wint-conversion]
29:   case 0x40051fULL: dest = *(volatile int8_t *)&ctx->ax; return 0;
29:                          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
29: /virtual/main.c:6:26: warning: incompatible integer to pointer conversion assigning to 'void *' from 'volatile int8_t' (aka 'volatile signed char') [-Wint-conversion]
29:   case 0x400537ULL: dest = *(volatile int8_t *)&ctx->ax; return 0;
29:                          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
29: /virtual/main.c:13:153: warning: incompatible integer to pointer conversion assigning to 'void *' from 'int32_t' (aka 'int') [-Wint-conversion]
29:   case 0x40051fULL: { u64 __addr = (*(volatile u64 *)&ctx->bp) + (-12);int32_t __res = 0x0; bpf_probe_read(&__res, sizeof(__res), (void *)__addr); dest = __res; } return 0;
29:                                                                                                                                                         ^ ~~~~~
29: /virtual/main.c:14:152: warning: incompatible integer to pointer conversion assigning to 'void *' from 'int32_t' (aka 'int') [-Wint-conversion]
29:   case 0x400537ULL: { u64 __addr = (*(volatile u64 *)&ctx->bp) + (-8);int32_t __res = 0x0; bpf_probe_read(&__res, sizeof(__res), (void *)__addr); dest = __res; } return 0;
29:                                                                                                                                                        ^ ~~~~~
29: /virtual/main.c:21:152: warning: incompatible integer to pointer conversion assigning to 'void *' from 'int32_t' (aka 'int') [-Wint-conversion]
29:   case 0x400520ULL: { u64 __addr = (*(volatile u64 *)&ctx->bp) + (-8);int32_t __res = 0x0; bpf_probe_read(&__res, sizeof(__res), (void *)__addr); dest = __res; } return 0;
29:                                                                                                                                                        ^ ~~~~~
29: /virtual/main.c:22:26: warning: incompatible integer to pointer conversion assigning to 'void *' from 'volatile int8_t' (aka 'volatile signed char') [-Wint-conversion]
29:   case 0x400545ULL: dest = *(volatile int8_t *)&ctx->ax; return 0;
29:                          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
29: /virtual/main.c:29:153: warning: incompatible integer to pointer conversion assigning to 'void *' from 'int32_t' (aka 'int') [-Wint-conversion]
29:   case 0x400520ULL: { u64 __addr = (*(volatile u64 *)&ctx->bp) + (-12);int32_t __res = 0x0; bpf_probe_read(&__res, sizeof(__res), (void *)__addr); dest = __res; } return 0;
29:                                                                                                                                                         ^ ~~~~~
29: /virtual/main.c:30:26: warning: incompatible integer to pointer conversion assigning to 'void *' from 'volatile int8_t' (aka 'volatile signed char') [-Wint-conversion]
29:   case 0x400545ULL: dest = *(volatile int8_t *)&ctx->dx; return 0;
29:                          ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
29: /virtual/main.c:36:8: warning: incompatible integer to pointer conversion assigning to 'void *' from 'int' [-Wint-conversion]
29:   dest = 5;
29:        ^ ~
29: /virtual/main.c:41:8: warning: incompatible integer to pointer conversion assigning to 'void *' from 'volatile int8_t' (aka 'volatile signed char') [-Wint-conversion]
29:   dest = *(volatile int8_t *)&ctx->ax;
29:        ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```